src/config/index.ts

```diff
@@ -95,3 +96,56 @@ export const buildSafeCreationTxUrl = (safeAddress: string) => {
 
   return `${host}${base}`
 }
+
+const fetchContractABI = memoize(
+  async (url: string, contractAddress: string, apiKey?: string) => {
+    let params: any = {
+      module: 'contract',
+      action: 'getAbi',
+      address: contractAddress,
+    }
+
+    if (apiKey) {
+      params = { ...params, apiKey }
+    }
+
+    const response = await fetch(`${url}?${new URLSearchParams(params)}`)
+
+    if (!response.ok) {
+      return { status: 0, result: [] }
+    }
+
+    return response.json()
+  },
+  (url, contractAddress) => `${url}_${contractAddress}`,
+)
+
+const getNetworkExplorerApiKey = (networkExplorerName: string): string | undefined=> {
+  switch (networkExplorerName) {
+    case 'Etherscan': {
+      return  ETHERSCAN_API_KEY
+    }
+    default: {
+      return undefined
+    }
+  }
```

Comment:
Can we normalize the value? I mean...

```typescript
switch (networkExplorerName.toLowerCase()) {
  case 'etherscan':
// ...
```

still, I'm having second thoughts about how effective this verification is, but it's ok so far.